lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -367,6 +375,50 @@ def show_inheritance_chain(
 
         return DatasetInheritanceChain(__root__=result)
 
+    def get_drs_object(
+        self,
+        trans: ProvidesHistoryContext,
+        object_id: EncodedDatabaseIdField,
```

Comment:
Since I suppose there'll be some pressure for the object_id to be stable: Should we use something like `hda:uuid` instead of the encoded database id ? That way we can serve objects that are not hdas in the future, and we avoid issues if someone rotates the id_secret